spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -149,6 +152,14 @@ public SparkInterpreter(Properties property, SparkContext sc) {
     this.sc = sc;
     env = SparkEnv.get();
     sparkListener = setupListeners(this.sc);
+
+    this.singleThreadedExecutorSerivce = createSingleThreadExecutorService();
+  }
+
+  private ExecutorService createSingleThreadExecutorService() {
+    return new ThreadPoolExecutor(1, 1, 0, TimeUnit.NANOSECONDS,
```

Comment:
Why not using `Executors.newSingleThreadExecutor()`

Reply:
because it creates `unbounded queue` and I though having an unbounded queue is bad thing.

Reply:
If I have an unbounded queue and If I use unbounded queue incorrectly and add huge amount of work to queue, I would notice that symptom only with memory usage.